cmd/access/cluster.go

```diff
@@ -0,0 +1,446 @@
+
+		input, err := osdctlutil.StreamRead(streams, '\n')
+		if err != nil {
+			return err
+		}
+		if !isAffirmative(strings.TrimSpace(input)) {
+			return fmt.Errorf("Did not impersonate '%s'", impersonateUser)
+		}
+		*flags.Impersonate = impersonateUser
+		osdctlutil.StreamPrintln(streams, fmt.Sprintf("Continuing as '%s'", impersonateUser))
+		osdctlutil.StreamPrintln(streams, "")
+	}
+	return nil
+}
+
+// clusterAccessOptions contains the objects and information required to access a cluster
+type clusterAccessOptions struct {
+	*genericclioptions.ConfigFlags
+	genericclioptions.IOStreams
+	kclient.Client
+}
+
+// newAccessOptions creates a clusterAccessOptions object
+func newClusterAccessOptions(client kclient.Client, streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) clusterAccessOptions {
+	a := clusterAccessOptions{
+		IOStreams:   streams,
+		ConfigFlags: flags,
+		Client:      client,
+	}
+	return a
+}
+
+// Println appends a newline then prints the given msg using the clusterAccessOptions' IOStreams
+func (c *clusterAccessOptions) Println(msg string) {
+	osdctlutil.StreamPrintln(c.IOStreams, msg)
+}
+
+// Print prints the given msg using the clusterAccessOptions' IOStreams
+func (c *clusterAccessOptions) Print(msg string) {
+	osdctlutil.StreamPrint(c.IOStreams, msg)
+}
+
+// Errorln appends a newline then prints the given error msg using the clusterAccessOptions' IOStreams
+func (c *clusterAccessOptions) Errorln(msg string) {
+	osdctlutil.StreamErrorln(c.IOStreams, msg)
+}
+
+// Readln reads a single line of user input using the clusterAccessOptions' IOStreams. User input is returned with all
+// procceeding and following whitespace trimmed
+func (c *clusterAccessOptions) Readln() (string, error) {
+	in, err := osdctlutil.StreamRead(c.IOStreams, '\n')
+	return strings.TrimSpace(in), err
+}
+
+// Run executes the 'cluster' access subcommand
+func (c *clusterAccessOptions) Run(cmd *cobra.Command, args []string) error {
+	clusterIdentifier := args[0]
+	c.Println(fmt.Sprintf("Retrieving Kubeconfig for cluster '%s'", clusterIdentifier))
+
+	// Connect to ocm
+	conn := osdctlutil.CreateConnection()
+	defer func() {
+		cmdutil.CheckErr(conn.Close())
+	}()
+
+	cluster, err := osdctlutil.GetCluster(conn, clusterIdentifier)
+	if err != nil {
+		return err
+	}
+	c.Println(fmt.Sprintf("Internal Cluster ID: %s", cluster.ID()))
+
+	// Retrieve the kubeconfig secret from the cluster's namespace on hive
+	ns, err := getClusterNamespace(c.Client, cluster.ID())
+	if err != nil {
+		return err
+	}
+	c.Println(fmt.Sprintf("Cluster namespace: %s", ns.Name))
+
+	kubeconfigSecret, err := c.getKubeConfigSecret(ns)
+	if err != nil {
+		return err
+	}
+	c.Println(fmt.Sprintf("Kubeconfig Secret: %s", kubeconfigSecret.Name))
+
+	if cluster.AWS().PrivateLink() {
+		c.Println("")
+		c.Println("Cluster is PrivateLink, and is only accessible via a jump pod on Hive")
+		return c.createJumpPodAccess(cluster, kubeconfigSecret)
+	} else {
```

Comment:
Same sorta thing I mention [below](https://github.com/openshift/osdctl/pull/250/files#r978965138): I think the structure of an `if-else` block makes sense in this context, since we're switching over the possible cluster access methods based on the properties of the clusters.

Does anyone else have thoughts on this?

Reply:
Based on the conversation below, it appears I've been outvoted on this - I've updated the logic to reflect these changes :+1: